cpp/CMakeLists.txt

```diff
@@ -390,6 +393,44 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/include/raft/version_config.hpp
         COMPONENT raft
         DESTINATION include/raft)
 
+##############################################################################
+# - export/install optional components  --------------------------------------
+
+include("${rapids-cmake-dir}/export/write_dependencies.cmake")
+
+set(raft_components distance nn)
+set(raft_install_comp raft raft)
+if(TARGET raft_distance_lib)
+  list(APPEND raft_components distance-lib)
+  list(APPEND raft_install_comp distance)
+endif()
+if(TARGET raft_nn_lib)
+  list(APPEND raft_components nn-lib)
+  list(APPEND raft_install_comp nn)
+endif()
+
+foreach(comp install_comp IN ZIP_LISTS raft_components raft_install_comp)
+  install(
+          EXPORT raft-${comp}-exports
+          FILE raft-${comp}-targets.cmake
+          NAMESPACE raft::
+          DESTINATION "${lib_dir}/cmake/raft"
+          COMPONENT raft${install_comp}
```

Comment:
With your last change to the `install_comp` names you'll also need to change this line and the second `rapids_export_write_dependencies` on line 429 below.

Reply:
Updated the install component names to be `nn` and `distance` so that the logic overall is easier to follow